bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -3877,6 +3848,47 @@ void ReplicaImp::executeRequestsInPrePrepareMsg(concordUtils::SpanWrapper &paren
   }
 }
 
+void ReplicaImp::executeRequestsAndSendResponses(IRequestsHandler::ExecutionRequestsQueue &accumulatedRequests,
+                                                 const std::string &batchCid,
+                                                 concordUtils::SpanWrapper &span) {
+  if (ReplicaConfig::instance().blockAccumulation) {
+    LOG_DEBUG(GL, "Executing all the requests of preprepare message with cid: " << batchCid << " With accumulation");
+    {
+      TimeRecorder scoped_timer(*histograms_.executeWriteRequest);
+      bftRequestsHandler_.execute(accumulatedRequests, batchCid, span);
+    }
+  } else {
+    LOG_DEBUG(GL, "Executing all the requests of preprepare message with cid: " << batchCid << " Without accumulation");
+    std::deque<IRequestsHandler::ExecutionRequest> singleRequest;
+    for (auto &req : accumulatedRequests) {
+      singleRequest.push_back(req);
+      {
+        TimeRecorder scoped_timer(*histograms_.executeWriteRequest);
+        bftRequestsHandler_.execute(singleRequest, batchCid, span);
+      }
+      req = singleRequest.at(0);
+      singleRequest.clear();
+    }
+  }
+  for (auto &req : accumulatedRequests) {
+    ConcordAssertGT(req.outActualReplySize,
+                    0);  // TODO(GG): TBD - how do we want to support empty replies? (actualReplyLength==0)
+    auto status = req.outExecutionStatus;
+    if (status != 0) {
+      const auto requestSeqNum = req.requestSequenceNum;
+      LOG_WARN(CNSUS, "Request execution failed: " << KVLOG(req.clientId, requestSeqNum));
```

Comment:
+cid? Add scoped log at the beginning of the function.

Reply:
Added.